protos/sources/data-node/api/v2/trading_data.proto

```diff
@@ -3870,3 +3885,93 @@ message GetReferralFeeStatsResponse {
+  // Volume discount program currently activated on the network.
+  VolumeDiscountProgram current_volume_discount_program = 1;
+}
+
+// Request to get the volume discount statistics for a given epoch.
+message GetVolumeDiscountStatsForEpochRequest {
+  // Optional epoch at which the statistics have been updated. If not set, last
+  // epoch is used.
+  uint64 at_epoch = 1;
+}
+
+// Response containing the volume discount statistics for the given epoch.
+message GetVolumeDiscountStatsForEpochResponse {
+  // Volume discount statistics for the epoch.
+  VolumeDiscountStatsForEpoch stats = 1;
+}
+
+// Request to get the volume discount statistics for a given party over the epochs.
+message GetVolumeDiscountStatsForPartyRequest {
+  // Party ID.
+  string party_id = 1;
+}
+
+// Response containing the volume discount statistics for the given party over the epochs.
+message GetVolumeDiscountStatsForPartyResponse {
+  // Volume discount statistics for the party.
+  VolumeDiscountStatsForParty stats = 1;
+}
+
+// Volume discount statistics for a given epoch.
+message VolumeDiscountStatsForEpoch {
+  // Epoch at which the statistics have been updated.
+  uint64 at_epoch = 1;
+  // Parties' statistics for the given epoch.
+  repeated PartyVolumeDiscountStats stats = 2;
+}
+
+message PartyVolumeDiscountStats {
+  // Party ID.
+  string party_id = 1;
+  // Discount factor applied to the volume.
+  string discount_factor = 2;
+  // The party's running volume.
+  string running_volume = 3;
+}
+
+// Volume discount statistics for a given party over the epochs.
+message VolumeDiscountStatsForParty {
+  // Party ID.
+  string party_id = 1;
+  // Party's statistics over the epochs.
+  repeated EpochVolumeDiscountStats stats = 2;
+}
+
+message EpochVolumeDiscountStats {
+  // Epoch at which the statistics have been updated.
+  uint64 at_epoch = 1;
+  // Discount factor applied to the volume.
+  string discount_factor = 2;
+  // The party's running volume.
+  string running_volume = 3;
+}
+
+// Volume discount program for reporting purposes differs slightly from the volume discount program
+// message in the volume discount program started and volume discount updated events, as it includes
+// an optional timestamp that is set when the program has ended.
+message VolumeDiscountProgram {
+  // Incremental version of the program. It is incremented after each program
+  // update.
+  uint64 version = 1;
+  // Unique ID generated from the proposal that created this program.
+  string id = 2;
+  // Defined benefit tiers in increasing order. First element will give Tier 1,
+  // second element will give Tier 2, and so on. Determines the level of
+  // benefit a party can expect based on performance criteria.
+  repeated vega.VolumeBenefitTier benefit_tiers = 3;
+  // Timestamp as Unix time in seconds after which when the current epoch ends, the
+  // program's status will become STATE_CLOSED and benefits will be disabled.
```

Comment:
Is this right? Asking based on this comment in a different PR - https://github.com/vegaprotocol/vega/pull/9550#discussion_r1335859626